app/test_main.py

```diff
@@ -1 +1,50 @@
-# write your code here
+from unittest import mock
+
+import pytest
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "url, mock_internet, mock_valid, expected",
+    [
+        (
+            "https://www.google.com",
+            True,
+            True,
+            "Accessible",
+        ),
+        (
+            "https://www.google.com",
+            False,
+            True,
+            "Not accessible",
+        ),
+        (
+            "https://www.google.com",
+            True,
+            False,
+            "Not accessible",
+        ),
+        (
+            "https://www.google.com",
+            False,
+            False,
+            "Not accessible",
+        ),
+    ],
```

Comment:
no need to write url here as param, u can pass it directly to can_access function